### PR TITLE
Migrate mocking code from openbare

### PR DESCRIPTION
**Pull request by smarlowucf**
- Use pytest instead of nose tests
- Cleanup license and author information

### CODE REVIEW

File: mockboto3/mockboto3/iam/models.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+
+""""Iam Classes."""
+
```
Comment (rjschwei) [inline mockboto3/mockboto3/iam/models.py]: IAM is an acronym, thus should be used in all CAPS in this case

**Comment by rjschwei** [comment]
mock [1] has a BSD license, why use MIT? Compatibility with some other code?

[1] https://pypi.python.org/pypi/mock

**Comment by smarlowucf** [comment]
It's also closely correlated with the boto3 library which is under Apache license. Should the license be determined to be compatible with required packages or best fit for library itself?

Note: That's a more general question. I'll bump this library to BSD. Hopefully it will become obsolete in the near future as Python3 support in the moto library is being actively fixed.

File: mockboto3/mockboto3/iam/constants.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    'MA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCpB/vsOwmT/O0td1RqzKjttSBaPjbr' \
+    'dqwNe9BrOyB08fw2+Ch5oonZYXfGUrT6mkYXH5fQot9HvASrzAKHO596FdJA6DmL' \
+    'ywdWe1Oggk7zFSXO1Xv+3vPrJtaYxYo3eRIp7w80PMkiOv6M0XK8ubcTouODeJbf' \
+    'suDqcLnLDxwsvwIDAQABo1cwVTAOBgNVHQ8BAf8EBAMCBaAwFgYDVR0lAQH/BAww' \
+    'CgYIKwYBBQUHAwIwDAYDVR0TAQH/BAIwADAdBgNVHQ4EFgQULGNaBphBumaKbDRK' \
+    'CAi0mH8B3mowDQYJKoZIhvcNAQEFBQADgYEAuKxhkXaCLGcqDuweKtO/AEw9ZePH' \
+    'wr0XqsaIK2HZboqruebXEGsojK4Ks0WzwgrEynuHJwTn760xe39rSqXWIOGrOBaX' \
+    'wFpWHVjTFMKk+tSDG1lssLHyYWWdFFU4AnejRGORJYNaRHgVTKjHphc5jEhHm0BX' \
+    'AEaHzTpmEXAMPLE=' \
+    '-----END CERTIFICATE-----'
```
Comment (rjschwei) [inline mockboto3/mockboto3/iam/constants.py:23]: Convention for constants is all UPPER_CASE

File: mockboto3/mockboto3/iam/endpoints.py [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+from mockboto3.core.utils import inflection
+
+from mockboto3.iam import responses
+from mockboto3.iam.models import AccessKey, Group, Policy, User
+from mockboto3.iam.utils import get_value_from_arn
+
+
+class MockIAM(object):
+    """Class for mocking iam endpoints."""
+
```
Comment (rjschwei) [inline mockboto3/mockboto3/iam/endpoints.py]: should be IAM in the docstring as well

File: mockboto3/mockboto3/iam/endpoints.py [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+        If the AWS endpoint is not mocked raise a client error.
+        """
+        try:
+            return getattr(self, inflection(operation_name))(kwarg)
+        except AttributeError:
+            raise client_error(operation_name,
+                               'NoSuchMethod',
+                               'Operation not mocked.')
+
+    def add_user_to_group(self, kwarg):
```
Comment (rjschwei) [inline mockboto3/mockboto3/iam/endpoints.py]: I think convention is to use kwargs not kwarg

File: mockboto3/mockboto3/iam/endpoints.py [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+            if kwarg['CertificateBody'] == cert.body:
+                raise client_error('UploadSigningCertificate',
+                                   'DuplicateCertificate',
+                                   'A duplicate certificate already exists.')
+
+        cert = user.upload_signing_certificate(kwarg['CertificateBody'])
+        return responses.upload_signing_certificate_response(kwarg['UserName'],
+                                                             cert)
+
+    def _check_user_exists(self, user, method):
```
Comment (rjschwei) [inline mockboto3/mockboto3/iam/endpoints.py]: In the absence of any logical closeness needs please sort methods in alpha order

File: mockboto3/tests/test_iam.py [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+import pytest
+
+from mockboto3.core.exceptions import MockBoto3ClientError
+from mockboto3.core.utils import inflection
+from mockboto3.iam.constants import policy_document, signing_cert
+from mockboto3.iam.endpoints import MockIAM, mock_iam
+
+
+class TestIam:
+
```
Comment (rjschwei) [inline mockboto3/tests/test_iam.py]: IAM

**Review by rjschwei — CHANGES_REQUESTED** [review]

**Review by rjschwei — APPROVED** [review]

**Pull request merged**